unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/BuilderMode/BuilderInWorldController.cs

```diff
@@ -321,17 +324,18 @@ public VoxelEntityHit GetCloserUnselectedVoxelEntityOnPointer()
                 }
             }
         }
+
         return voxelEntityHit;
     }
 
-    private void NewSceneAdded(ParcelScene newScene)
+    private void NewSceneAdded(IParcelScene newScene)
     {
         if (newScene.sceneData.id != sceneToEditId)
             return;
 
         Environment.i.world.sceneController.OnNewSceneAdded -= NewSceneAdded;
 
-        sceneToEdit = Environment.i.world.state.GetScene(sceneToEditId) as ParcelScene;
+        sceneToEdit = (ParcelScene)Environment.i.world.state.GetScene(sceneToEditId);
```

Comment:
I'm a bit concerned of all these castings we have to make regarding the scenes. It feels like something is missing somewhere, as if our extracted interfaces for the scenes were a bit incompleted.

Reply:
This was another shortcut. `BuilderInWorldController` needs the `ParcelScene` type deep down in the logic to call the scene messages manually. Ideally, it should use some kind of interface to do this because messages processing shouldn't belong in `ParcelScene`.

Ty for reminding me of this issue, will add it as tech debt because the fix is not trivial.